java-frontend/src/test/java/org/sonar/java/cfg/CFGLoopTest.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.cfg;
+
+import org.junit.Test;
+import org.sonar.java.cfg.CFG.Block;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+public class CFGLoopTest {
+
+  private static final CFGTestLoader loader = new CFGTestLoader("src/test/files/cfg/CFGLoopTest.java");
+
+  @Test
+  public void simpleWhileLoop() {
+    final CFG cfg = CFG.build(loader.getMethod("CFGLoopTest", "simpleWhileLoop"));
+    Map<Tree, CFGLoop> loops = CFGLoop.getCFGLoops(cfg);
+    assertThat(loops).hasSize(1);
+    CFGLoop loop = loops.values().iterator().next();
+    List<Block> blocks = CFGLoop.sorted(cfg.blocks());
+    assertThat(loop.startingBlock()).isSameAs(blocks.get(2));
+    assertThat(loop.blocks()).containsOnly(blocks.get(1));
+    assertThat(loop.hasNoWayOut()).isTrue();
+  }
+
+  @Test
+  public void simpleWhileLoopWithBreak() {
+    final CFG cfg = CFG.build(loader.getMethod("CFGLoopTest", "simpleWhileLoopWithBreak"));
+    Map<Tree, CFGLoop> loops = CFGLoop.getCFGLoops(cfg);
+    assertThat(loops).hasSize(1);
+    CFGLoop loop = loops.values().iterator().next();
+    List<Block> blocks = CFGLoop.sorted(cfg.blocks());
+    assertThat(loop.startingBlock()).isSameAs(blocks.get(3));
+    assertThat(loop.blocks()).containsOnly(blocks.get(2), blocks.get(1));
```

Comment:
Be careful, the `containsOnly` assertion does not check for the order. I think that it this context, the order is important.